diff -ru SDL_net-1.2.7/SDLnetsys.h SDL_net-1.2.7-TCP_NODELAY/SDLnetsys.h
--- SDL_net-1.2.7/SDLnetsys.h	2009-05-28 11:25:58.000000000 -0400
+++ SDL_net-1.2.7-TCP_NODELAY/SDLnetsys.h	2009-05-28 11:33:21.000000000 -0400
@@ -55,10 +55,7 @@
 #ifndef __BEOS__
 #include <arpa/inet.h>
 #endif
-#if defined(linux) || defined(MACOS_OPENTRANSPORT)  /* FIXME: what other platforms have this? */
-#warning "Using netinet/tcp.h"
 #include <netinet/tcp.h>
-#endif
 #include <netdb.h>
 #include <sys/socket.h>
 #endif /* WIN32 */
