diff -ru curl-7.19.4/lib/transfer.c curl-7.19.4-EINTR/lib/transfer.c
--- curl-7.19.4/lib/transfer.c	Fri Feb 27 02:53:37 2009
+++ curl-7.19.4-EINTR/lib/transfer.c	Tue May 26 15:29:06 2009
@@ -1873,12 +1873,10 @@
 
     switch (Curl_socket_ready(fd_read, fd_write, timeout_ms)) {
     case -1: /* select() error, stop reading */
-#ifdef EINTR
       /* The EINTR is not serious, and it seems you might get this more
          often when using the lib in a multi-threaded environment! */
       if(SOCKERRNO == EINTR)
         continue;
-#endif
       return CURLE_RECV_ERROR;  /* indicate a network problem */
     case 0:  /* timeout */
     default: /* readable descriptors */
